cmake/extensions.cmake

```diff
@@ -2245,3 +2279,379 @@ function(target_byproducts)
                      COMMENT "Logical command for additional byproducts on target: ${TB_TARGET}"
   )
 endfunction()
+
+########################################################
```

Comment:
I think we should put these in their own file (cmake/linker.cmake) and think about removing the `zephyr_` prefix as I think this could be used by other projects.

Reply:
makes sense.

Reply:
@galak, not sure if `linker.cmake` is best choice directly at the `<zephyr>/cmake/` level as it may not be obvious to people that this particular file contains CMake extension functions / macros.

Maybe we should refactor and the create:
```